### DISCUSSION TITLE
Ideas for a Test Media Library

### DISCUSSION THREAD

**Discussion by PrplHaz4** [jellyfin/jellyfin-meta#49] [Ideas]
This is a quick dump based on a Matrix session...

**Goal**
Create a media library that can be used for testing Jellyfin in a repeatable, and eventually automated manner.

**Distribution Options**
1. Git Repo
2. Downloader script
3. Downloader script + FFMPEG to get desired formats
4. Downloadable archive from Jellyfin Web Server

**Supported Test Cases**
1. Add/Browse/Scan Library
2. Play Media
3. Subtitle Support
4. Library Permissions
5. DLNA Media Server

**Sources**
1. GNOME Sample Library [Github](https://github.com/GNOME/gupnp-dlna-media)
Audio, video, images, multiple formats/quality
2. jell.yfish.us Test Media [jell.yfish.us](http://jell.yfish.us/)
H.264 and H.265/HEVC encoded .mkv video clips that can be used for testing the network streaming and playback performance of media streamers & HTPCs
3. Internet Archive Feature Films [Internet Archive](https://archive.org/details/feature_films)
4. Blender Open Projects [Blender](https://www.blender.org/about/projects/)
5. Public Domain Movies [Link](http://publicdomainmovies.net/feature_movies)
6. FFmpeg Samples [FFmpeg](https://samples.ffmpeg.org/)
7. NASA Downloadable Videos [NASA](https://www.nasa.gov/multimedia/downloadable-video-page/)

**Comment by Bond-009** [comment]
[FFmpeg samples](https://samples.ffmpeg.org/)

**Comment by Artiume** [comment]
Also NASA videos ;)

**Comment by PrplHaz4** [comment]
> 
> 
> Also NASA videos ;)

Admittedly didn't look too hard, but I found most of the NASA video links were to tube sites - no plain video files.

My initial criteria for sample files was to be sure to include stuff that would appear in TMDB or TVDB to facilitate scanning tests - but I'm realizing that should probably only be a subset...

**Comment by Artiume** [comment]
https://www.nasa.gov/multimedia/guidelines/index.html

Did a little digging and it looks like they host the majority of everything on YT. 
https://www.nasa.gov/mission_pages/station/videos/index.html


I did find this
https://www.nasa.gov/multimedia/downloadable-video-page/

https://mars.nasa.gov/multimedia/videos/?page=0&per_page=25&order=pub_date+desc&search=&condition_1=1%3Ais_in_resource_list&category=53

Yeah, none of these would have a TM/TVDB connection lol. It could be a good sample for 'Mixed Content' perhaps

**Comment by Bond-009** [comment]
UVG Dataset
http://ultravideo.fi/#testsequences_x

**Comment by Vyoam** [comment]
I had to dig a bit to find one with subtitles...
https://commons.wikimedia.org/wiki/File:Sprite_Fright_-_Blender_Open_Movie-full_movie.webm
Well, the subtitles aren't embedded, but good enough...
https://commons.wikimedia.org/wiki/TimedText:Sprite_Fright_-_Blender_Open_Movie-full_movie.webm.en.srt

**Comment by Chaphasilor** [comment]
Please don't forget to add some music for testing :)  
Or any audio with metadata 😅
